projects/ng7-mat-breadcrumb/src/lib/ng7-mat-breadcrumb.component.ts

```diff
@@ -22,40 +26,49 @@ export class Ng7MatBreadcrumbComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.ng7MatBreadcrumbService.breadcrumbLabels.subscribe((labelData) => {
+    this.ng7MatBreadcrumbService.breadcrumbLabels.subscribe(labelData => {
       for (const label in labelData) {
         if (labelData.hasOwnProperty(label)) {
-          this.breadcrumb.map((crumb) => {
+          this.breadcrumb.map(crumb => {
             const labelParams = crumb.label.match(/[^{{]+(?=\}})/g);
             if (labelParams) {
               for (const labelParam of labelParams) {
-                const dyanmicData = labelData[label];
+                const dynamicData = labelData[label];
                 if (labelParam === label) {
-                  crumb.label = crumb.label.replace('{{' + labelParam + '}}', dyanmicData);
+                  crumb.displayText = crumb.label.replace(
+                    '{{' + labelParam + '}}',
+                    dynamicData
+                  );
```

Comment:
Instead of overwriting the label, set the displayText and leave the label intact.